Fix TODO - bot names

jake-87 opened the pull request:
Added GH-style default names to the bots, using a list of animals.

jake-87 committed:
Fix TODO - bot names

Added GH-style default names to the bots, using a list of animals.

joelgibson commented on client/necsus.js:
```diff
@@ -219,9 +219,26 @@ let app = new Vue({
+                       "Perch", "Pheasant", "Pig", "Pigeon", "Porcupine", "Quagga", "Rabbit", "Raccoon", "Rat",
+                       "Rattlesnake", "Rooster", "Seal", "Sheep", "Skunk", "Sloth", "Snail", "Snake", "Spider",
+                       "Tiger", "Whale", "Wolf", "Wombat", "Zebra"]
+      const shuffled = animals.sort(() => 0.5 - Math.random());
```
Sorting using a random comparator is not quite the same thing as shuffling, and doesn't make much sense (the sorting algorithm could ask if `x <= y` and get `true`, and later ask the same question and get `false`). How you can get a random shuffle via sorting is by generating a random number for each element of the list, hence turning every item into a pair of `(data, random number)`, and sorting by the random number, like in this answer: <https://stackoverflow.com/a/46545530>

joelgibson reviewed (commented):
Hi Jake, thanks for the PR! If you can change your shuffling algorithm to something which is genuinely a shuffle (or some other way of choosing three items at random without replacement), then I'll happily merge it.

jake-87 committed:
Fix sort to use an actual sorting algorithm

Previous version used an ad-hoc sorting mechanism, fixed to use a proper mechanism based on `Math.random`.

jake-87 commented:
Should be good to go now. Apologies for mostly copying directly from SO, but it seemed like the most reasonable way to do it.

jake-87 commented:
Enjoy NCSS 2024 ;)
